python/lsst/afw/image/imagePca.cc

```diff
@@ -36,51 +37,59 @@ namespace image {
-    py::class_<ImagePca<ImageT>, std::shared_ptr<ImagePca<ImageT>>> cls(mod, ("ImagePca" + suffix).c_str());
+static void declareImagePca(lsst::utils::python::WrapperCollection &wrappers, std::string const &suffix) {
+    std::string name="ImagePca" + suffix;
+    wrappers.wrapType( py::class_<ImagePca<ImageT>,std::shared_ptr<ImagePca<ImageT>>>(wrappers.module, name.c_str()), [](auto &mod, auto &cls) {
+        //  py::class_<ImagePca<ImageT>, std::shared_ptr<ImagePca<ImageT>>> cls(mod, ("ImagePca" + suffix).c_str());
 
-    cls.def(py::init<bool>(), "constantWeight"_a = true);
+        cls.def(py::init<bool>(), "constantWeight"_a = true);
 
-    cls.def("addImage", &ImagePca<ImageT>::addImage, "img"_a, "flux"_a = 0.0);
-    cls.def("getImageList", &ImagePca<ImageT>::getImageList);
-    cls.def("getDimensions", &ImagePca<ImageT>::getDimensions);
-    cls.def("getMean", &ImagePca<ImageT>::getMean);
-    cls.def("analyze", &ImagePca<ImageT>::analyze);
-    cls.def("updateBadPixels", &ImagePca<ImageT>::updateBadPixels);
-    cls.def("getEigenValues", &ImagePca<ImageT>::getEigenValues);
-    cls.def("getEigenImages", &ImagePca<ImageT>::getEigenImages);
+        cls.def("addImage", &ImagePca<ImageT>::addImage, "img"_a, "flux"_a = 0.0);
+        cls.def("getImageList", &ImagePca<ImageT>::getImageList);
+        cls.def("getDimensions", &ImagePca<ImageT>::getDimensions);
+        cls.def("getMean", &ImagePca<ImageT>::getMean);
+        cls.def("analyze", &ImagePca<ImageT>::analyze);
+        cls.def("updateBadPixels", &ImagePca<ImageT>::updateBadPixels);
+        cls.def("getEigenValues", &ImagePca<ImageT>::getEigenValues);
+        cls.def("getEigenImages", &ImagePca<ImageT>::getEigenImages);
+    });
 }
 
 template <typename Image1T, typename Image2T>
-static void declareInnerProduct(py::module &mod) {
-    mod.def("innerProduct",
-            (double (*)(Image1T const &, Image2T const &, int const))innerProduct<Image1T, Image2T>, "lhs"_a,
-            "rhs"_a, "border"_a = 0);
+static void declareInnerProduct(lsst::utils::python::WrapperCollection &wrappers) {
+    wrappers.wrap([](auto &mod) {
+        mod.def("innerProduct",
+                (double (*)(Image1T const &, Image2T const &, int const)) innerProduct<Image1T, Image2T>, "lhs"_a,
+                "rhs"_a, "border"_a = 0);
+    });
 }
 
 }  // namespace
 
+
 PYBIND11_MODULE(imagePca, mod) {
-    declareImagePca<Image<int>>(mod, "I");
-    declareImagePca<Image<float>>(mod, "F");
-    declareImagePca<Image<double>>(mod, "D");
-    declareImagePca<Image<std::uint16_t>>(mod, "U");
-    declareImagePca<Image<std::uint64_t>>(mod, "L");
-    declareImagePca<MaskedImage<int>>(mod, "MI");
-    declareImagePca<MaskedImage<float>>(mod, "MF");
-    declareImagePca<MaskedImage<double>>(mod, "MD");
-    declareImagePca<MaskedImage<std::uint16_t>>(mod, "MU");
-    declareImagePca<MaskedImage<std::uint64_t>>(mod, "ML");
+    lsst::utils::python::WrapperCollection wrappers(mod, "lsst.afw.image.imagePca");
```

Comment:
Needs some kind of dependency on `lsst.afw.image.image`, unless they're going to be linked together shortly.

Reply:
image ticket.